inbm/dispatcher-agent/dispatcher/source/ubuntu_source_manager.py

```diff
@@ -98,8 +97,10 @@ def __init__(self) -> None:
         pass
 
     def add(self, parameters: ApplicationAddSourceParameters) -> None:
-        # Step 1: Add key
-        add_gpg_key(parameters.gpg_key_uri, parameters.gpg_key_name)
+        """Adds a source file and optional GPG key to be used during Ubuntu application updates."""
+        # Step 1: Add key (Optional)
+        if parameters.gpg_key_name and parameters.gpg_key_uri:
```

Comment:
Consider logging an informative message when GPG key parameters are not provided, for better traceability and debugging. [medium]